### DISCUSSION TITLE
Setting the seed for replication purposes

### DISCUSSION THREAD

**Discussion by AdamBaumann** [DoubleML/doubleml-for-r#184] [Q&A]
When constructing a DoubleMLPLR, I want to make sure that the sample splitting is such that exact replication of the estimates will be possible. Can a seed be set within the package such that the results are the same for each run, taking into account the operating system and such? If not, do you have any other recommendations for achieving this when using the package?

**Comment by PhilippBach** [comment]
Hi @AdamBaumann, 

thanks for that issue. We don't have a internal implementation on that as we generally use R's `random.seed()`.  I'm not an expert on how well that works across different operating systems, but I think this is a very general issue to R and maybe there is a solution, see for example https://stackoverflow.com/questions/64302978/set-seed-over-different-os-with-rngkind . 

Please note that there is also a second layer of randomness involved which is related to the employed learners. For example, we realized that due to randomness of some learning algorithms (like random forests, which are based on resampling and random variable choice) it is basically impossible (at least it doesn't work through `set.seed()`) to align results across platforms; at the time we were facing this issue, we think it was due to the underlying C++ implementation. However, the numerical differences were very small . Just wanted to point at this randomness, too, just in case you get the sample split aligned and wonder about additional differences

Another alternative would be to specify the sample split indices externally and use them accordingly, see https://docs.doubleml.org/stable/guide/resampling.html#externally-provide-a-sample-splitting-partition
